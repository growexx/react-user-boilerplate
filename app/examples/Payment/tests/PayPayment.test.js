import React from 'react';
import { render } from 'react-testing-library';
import PayPayment from '../PayPayment';

const props = {};

const componentWrapper = updatedProps =>
  render(<PayPayment {...props} {...updatedProps} />);

describe('<PayPayment/>', () => {
  it('should render with no props', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with props type paypal', () => {
    const {
      container: { firstChild },
    } = componentWrapper({
      type: 'Paypal',
      amount: '10',
    });
    expect(firstChild).toMatchSnapshot();
  });
  it('should render with props type paytm', () => {
    const {
      container: { firstChild },
    } = componentWrapper({
      type: 'Paytm',
      amount: '10',
    });
    expect(firstChild).toMatchSnapshot();
  });
});
