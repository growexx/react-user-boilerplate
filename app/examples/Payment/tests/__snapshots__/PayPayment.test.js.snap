// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PayPayment/> should render with no props 1`] = `null`;

exports[`<PayPayment/> should render with props type paypal 1`] = `
<div
  class="logo"
>
  <img
    alt="paypal_logo"
    src="IMAGE_MOCK"
  />
  <div>
    <button
      class="ant-btn pay_btn"
      type="button"
    >
      <span>
        Pay Now ₹ 10
      </span>
    </button>
  </div>
</div>
`;

exports[`<PayPayment/> should render with props type paytm 1`] = `
<div>
  <img
    alt="paytm_logo"
    class="paytm_logo"
    src="IMAGE_MOCK"
  />
  <div>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        Pay Now ₹ 10
      </span>
    </button>
  </div>
</div>
`;
